### PR TITLE
Add a name baseline

### PR DESCRIPTION
**Pull request by xsyann**
Add a new `Baseline` setting allowing to set a baseline with a smaller font size under the user name.

When provided:

Screenshot 2022-09-06 at 16 19 19
Screenshot 2022-09-06 at 16 19 10

When not provided:

Screenshot 2022-09-06 at 16 19 32
Screenshot 2022-09-06 at 16 19 38

### CODE REVIEW

File: core/client/components/character-name-text.js [hunk 1]
```diff
@@ -1,34 +1,85 @@
...
+  fontSize: 12,
+};
+
 const defaultTint = 0xFFFFFF;
+const baselineOffset = 20;
 
-class CharacterNameText extends Phaser.GameObjects.Text {
-  constructor(scene, text, tintName) {
-    super(scene, 0, 0, text);
+class CharacterNameText {
```
Comment (Donorhan) [inline core/client/components/character-name-text.js:18]: I wonder if we should not inherit from a `Phaser.GameObjects.Container` to have a "group" and have a relative transformation matrix.

File: core/server/users.js [hunk 2]
```diff
@@ -137,6 +137,7 @@ Meteor.methods({
       bio: Match.Optional(String),
       defaultReaction: Match.Optional(String),
       name: Match.Optional(String),
+      baseline: Match.Optional(String),
```
Comment (Donorhan) [inline core/server/users.js:140]: Not really related to these changes but I think we're going to have to start watching the entries, at least string length

File: core/client/components/character-name-text.js [hunk 3]
```diff
@@ -1,34 +1,85 @@
...
+    this.scene.add.existing(text);
+    return text;
+  }
+
+  setBaseline(text) {
+    if (!text) {
+      this.destroyBaseline();
+      return;
+    }
+    if (this.baseline) {
```
Comment (Donorhan) [inline core/client/components/character-name-text.js:43]: In general I prefer this syntax but it's fine the way you did it.

```
if (!this.baseline) this.baseline = this.createText(text, baselineStyle);
this.baseline.setText(text, baselineStyle);
```

File: core/server/users.js (hunk 2 above)
Comment (xsyann) [reply core/server/users.js:140]: Yes, I agree :)

File: core/client/components/character-name-text.js (hunk 1 above)
Comment (xsyann) [reply core/client/components/character-name-text.js]: Thanks for the suggestion, you're right it is clearly better :)

**Comment by xsyann** [comment]
> That's great, thanks!
> 
> But I'm afraid for the performance, the text rendering is something expensive for the GPU. If we want to avoid performance problems at rendering level, we'll have to use BitmapText (on the other hand, special characters can cause problems with BitmapText).
> 
> For the moment let's do it like this, we will improvise according to the feedback 🙂

Thanks for the feedback @Donorhan :)
I tried to do some tests using metrics (https://photonstorm.github.io/phaser3-docs/Phaser.Types.GameObjects.Text.html#.TextMetrics) to improve text rendering performances but unfortunately I didn't measure any improvement.

I tried to store metrics of the first rendered text and then loop to create hundreds of texts, but the time measured with or without metrics was similar.

**Comment by Donorhan** [comment]
@xsyann Thanks for your research, I think it will be visible on the GPU side with the number of drawcalls but I have the impression that Phaser handles it all pretty well internally 🙂

**Pull request merged**